### DISCUSSION TITLE
Search function not invoked when search text cleared?

### DISCUSSION THREAD

**Discussion by jamesmorgan** [algolia/instantsearch#4978] [Q&A] [answered]
Hi, I have vue component wired up and working as expected except when clearing the search text.

I am using the `ais-instant-search` component. 

```
    <ais-instant-search :search-client="searchClient"
                        :search-function="searchFunction"
                        index-name="my-index">
    </ais-instant-search>

   searchFunction(helper) {
        console.log('searchFunction', helper.state.query);
        helper
          .setQuery(helper.state.query)
          .search();
   },
```                    

The problem I am facing is that `searchFunction` is being invoked fine but when I remove the last character is cleared it is not? Is there a way to reset search when clear the search box completely e.g. `helper.state.query` will be empty?

**Comment by Haroenv** [comment, marked as answer]
Can you explain what you are trying to achieve? I don't fully understand the issue

**Reply by jamesmorgan** [reply]
When a users deletes the search term I want the search results to be clearer either to nothing or to the original default list for that index. Atm when the users clears the search term it does not reset search. How do I clear the search results when someone clears the search box?

**Reply by Haroenv** [reply]
That's indeed not the default behaviour of the search box reset, but can be worked around by using clear-refinements: https://codesandbox.io/s/affectionate-pond-pgd5h?file=/src/App.vue

**Reply by jamesmorgan** [reply]
With clea refinements does this mean the user has to click a button to trigger the clear refinements?

**Reply by Haroenv** [reply]
yes, did you mean when a user manually clears the input it should also clear the refinements? That's probably easier to do in onSearchStateChange, but I'm not sure if that's a good behaviour

**Reply by jamesmorgan** [reply]
I dont want to clear refinements just reset the freetext search box - atm this isnt happening when the users clears the entered search term. I will look at `onSearchStateChange` and see if this fits the bill

**Reply by Haroenv** [reply]
do you have a sandbox with the current behaviour? is this possibly fixed by https://github.com/algolia/instantsearch.js/pull/4990?

**Reply by jamesmorgan** [reply]
That does infact fix it! bumping to the latest worked!
